crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -601,23 +598,98 @@ mod tests {
     }
 
     #[test]
-    fn observer_multiple_events() {
+    fn observer_multiple_events_static() {
+        let mut world = World::new();
+        world.init_resource::<R>();
+        #[derive(Event)]
+        struct Foo(i32);
+        #[derive(Event)]
+        struct Bar(bool);
+        world.observe(|t: Trigger<(Foo, Bar)>, mut res: ResMut<R>| {
+            res.0 += 1;
+            match t.event() {
+                Or2::A(Foo(v)) => assert_eq!(5, *v),
```

Comment:
I like what the PR enables, but I think there are some ergonomic papercuts that should be resolved. It's not great that the user has to remember to use `Or2`, `Or3`, etc.

Reply:
Yea, unfortunately I don't think we're getting anything like anonymous enums anytime soon. Suggestions welcome on how to improve it!